python/ray/tune/trainable/function_trainable.py

```diff
@@ -454,64 +453,57 @@ def step(self):
     def execute(self, fn):
         return fn(self)
 
+    def get_state(self):
+        state = super().get_state()
+
+        checkpoint = self._status_reporter.get_checkpoint()
+        if not checkpoint:
+            state.update(iteration=0, timesteps_total=0, episodes_total=0)
+        return state
+
     def save_checkpoint(self, tmp_checkpoint_dir: str = ""):
         if tmp_checkpoint_dir:
             raise ValueError("Checkpoint dir should not be used with function API.")
 
         checkpoint = self._status_reporter.get_checkpoint()
-        state = self.get_state()
 
         if not checkpoint:
-            state.update(iteration=0, timesteps_total=0, episodes_total=0)
             # We drop a marker here to indicate that the checkpoint is empty
             checkpoint = FuncCheckpointUtil.mk_null_checkpoint_dir(self.logdir)
             parent_dir = checkpoint
         elif isinstance(checkpoint, dict):
-            parent_dir = TrainableUtil.make_checkpoint_dir(
```

Comment:
if i am reading the code correctly, `make_checkpoint_dir ` is not returning anything now?